### ISSUE TITLE
Longpix boosts

### ISSUE THREAD

**Issue by elswyyr** [eternaldensity/Sandcastle-Builder#1402]
According to the reddit post and my own testing, you can no longer unlock Doublepost and the other longpix boosts (except Overcompensating for some reason) from QQs, possibly not at all.

**Comment by EPSIL0N** [comment]
Overcompensating is simply unlocked while others are put into the department random choice pool, this would most likely also effect the JD boosts as someone else has reported as they go in the same pool.

I swear we handled this but I may have only tested with boosts from the logic pool and it is entirely likely that the entire testing team is past the DorD stage and the new reward handling that enables Mario to go magnitudes faster is interfering here.

Or it might be that there is a glitch from adding the kitten only choice pool.

**Comment by elswyyr** [comment]
There's a possibly related (but minor) bug with Robotic Shopper; when it's set to autobuy Vault Keys, you won't get Vault Keys from manually opening Question Qubes.

**Comment by EPSIL0N** [comment]
> There's a possibly related (but minor) bug with Robotic Shopper; when it's set to autobuy Vault Keys, you won't get Vault Keys from manually opening Question Qubes.

I would have to open up the particular save and watch a handful of variables, but it should simply be acting as if the key was unlocked, and then bought. As far as I can tell we did contain the single key scenario but have left it silenced. Again as part of enabling Mario to open magnitudes of QQs as silently and quickly as possible.

**Comment by elswyyr** [comment]
I've been watching the Locked Vault boost, and the amount of keys needed doesn't change, and neither does the amount of vaults opened.

**Comment by EPSIL0N** [comment]
I can not replicate the vault key issue as it currently works as expected for me. Could you pastebin a save and link to it?

The longpix and judgement dip issue are something to do with Molpy.BuildRewardsLists() not being called so the department list is not up to date.

**Comment by EPSIL0N** [comment]
I believe #1403 will address the issue with the longpix boosts and judgement dip boosts.

**Comment by elswyyr** [comment]
Here you go. http://pastebin.com/VVHUHSvz

**Comment by EPSIL0N** [comment]
Thanks. Well I found the issue with Rob and Vault keys and am not sure how to fix it.

In the short term just don't tell Rob to buy Vault keys as the problem is that we put a check for true unlimited castles from 'Aleph One' or 'Between the Cracks' when buying vault keys or vaults with Rob and this raises a new problem in that Mario is feasible to unlock before 'Aleph One'.

Currently it fails and they get nothing and anything is better than nothing so should I just have it reward up to one full vault per mNP if people don't have an infinity castle boost active?

**Comment by pickten** [comment]
What I previously believed it did was to default, like most other parts of that function, to `Molpy.UnlockBoost` in that case (and hence unlock 1), but I agree that going up to just under 2 is a better idea.

**Comment by EPSIL0N** [comment]
> What I previously believed it did was to default, like most other parts of that function, to Molpy.UnlockBoost in that case (and hence unlock 1)

Yeah, this was the one case we hadn't considered, it only defaults to just unlocking 1 for boosts that Rob has been told not to buy that are not free and possibly when being told to buy 'Locked Vault' without Aleph One but that almost never happens. I think I made some more improvements now.

**Comment by EPSIL0N** [comment]
Was told about a few and then found a few more boosts that didn't seem to be working for request #1414 , also looks like there will probably be more work to do in the future.
